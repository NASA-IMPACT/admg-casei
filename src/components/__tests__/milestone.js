import React from "react"
import renderer from "react-test-renderer"
import Milestone from "../timeline/milestone"

jest.mock("../image.js", () => {
  const mockImage = () => <img />
  return mockImage
})

describe("MileStone", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Milestone
          key="f2c42b8a-2c06-407c-9345-aa9824ac65f5"
          type="deployment"
          daterange="10/02/2012 - 10/20/2012"
          name="Canada, Oregon Fall 2012"
          details="2 Flights"
          region="Canada"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
