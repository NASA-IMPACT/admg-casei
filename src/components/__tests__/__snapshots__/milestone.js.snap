// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MileStone renders correctly 1`] = `
<div
  data-cy="milestone"
  style={
    Object {
      "padding": "3rem",
    }
  }
>
  <label
    style={
      Object {
        "backgroundColor": "#F93C21",
        "color": "#0c1520",
        "fontSize": "small",
        "left": "4rem",
        "padding": ".25rem",
        "position": "absolute",
        "textTransform": "uppercase",
        "top": "4rem",
      }
    }
  >
    deployment
  </label>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "flex": "1",
        }
      }
    >
      <img
        alt="Milestone-image"
        data-cy="overview-map"
        src="https://picsum.photos/300/300"
      />
    </div>
    <div
      style={
        Object {
          "flex": "1.61803398875",
          "padding": "1rem",
        }
      }
    >
      <label
        data-cy="timeline-milestone-date-label"
        style={
          Object {
            "borderBottom": "none",
            "color": "#9E9E9E",
            "display": "block",
            "fontSize": "small",
            "textTransform": "uppercase",
          }
        }
      >
        2012-10-02
         â€” 
        2012-10-20
      </label>
      <h3>
        Canada, Oregon Fall 2012
      </h3>
      <p>
        2 Flights
      </p>
      <p>
        Canada
      </p>
    </div>
  </div>
</div>
`;
